cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1654,19 +1672,131 @@ String zkConnectionString(int connectToReplicas)  {
                         this.kafkaService = kafkaCluster.generateService();
                         this.kafkaHeadlessService = kafkaCluster.generateHeadlessService();
 
                         return Future.succeededFuture(this);
                     });
         }
 
+        protected Future<Map<ConfigResource, Config>> getCurrentConfig(int podId, Admin ac) {
+            ConfigResource resource = new ConfigResource(ConfigResource.Type.BROKER, String.valueOf(podId));
+            return kafkaFutureToVertxFuture(ac.describeConfigs(singletonList(resource)).all());
+        }
+
+        protected Future<Admin> adminClient(AdminClientProvider adminClientProvider, String namespace, String cluster, int podId) {
+            Promise<Admin> acPromise = Promise.promise();
+            Future<Secret> clusterCaCertSecretFuture = secretOperations.getAsync(
+                    namespace, KafkaResources.clusterCaCertificateSecretName(cluster));
+            Future<Secret> coKeySecretFuture = secretOperations.getAsync(
+                    namespace, ClusterOperator.secretName(cluster));
+            String hostname = KafkaCluster.podDnsName(namespace, cluster, KafkaCluster.kafkaPodName(cluster, podId)) + ":" + KafkaCluster.REPLICATION_PORT;
+
+            return CompositeFuture.join(clusterCaCertSecretFuture, coKeySecretFuture).compose(compositeFuture -> {
+                Secret clusterCaCertSecret = compositeFuture.resultAt(0);
+                if (clusterCaCertSecret == null) {
+                    return Future.failedFuture(Util.missingSecretException(namespace, KafkaCluster.clusterCaCertSecretName(cluster)));
+                }
+                Secret coKeySecret = compositeFuture.resultAt(1);
+                if (coKeySecret == null) {
+                    return Future.failedFuture(Util.missingSecretException(namespace, ClusterOperator.secretName(cluster)));
+                }
+
+                Admin ac;
+                try {
+                    ac = adminClientProvider.createAdminClient(hostname, clusterCaCertSecret, coKeySecret, "cluster-operator");
+                    acPromise.complete(ac);
+                } catch (RuntimeException e) {
+                    log.warn("Failed to create Admin Client. {}", e);
+                    acPromise.complete(null);
+                }
+                return acPromise.future();
+            });
+        }
+
+        /**
+         * If the kafka pod is running, change its configuration dynamically if it is possible (and needed).
+         * If it is not possible set the flag for specific podID an it will be rolled in {@code kafkaRollingUpdate}.
+         * @return Succeeded future
+         */
+        Future<ReconciliationState> kafkaBrokerDynamicConfiguration() {
+            // patches scale up/down situations
+            int runningReplicas = Math.min(oldKafkaReplicas, kafkaCluster.getReplicas());
```

Comment:
This looks suspitios. are you really loooking for running replicas before reconciliation or after reocnciliation? I wonder if it is just badly chosen variable name (in which case you should rename it) or if it is some bug in the code.